terracotta/drivers/raster_base.py

```diff
@@ -479,24 +429,43 @@ def _get_raster_tile(cls, path: str, *,
             except OSError:
                 raise IOError('error while reading file {}'.format(path))
 
-            # compute suggested resolution and bounds in target CRS
-            dst_transform, _, _ = cls._calculate_default_transform(
+            # compute buonds in target CRS
+            dst_bounds = warp.transform_bounds(src.crs, cls._TARGET_CRS, *src.bounds)
+
+            if tile_bounds is None:
+                tile_bounds = dst_bounds
+
+            # prevent loads of very sparse data
+            cover_ratio = (
+                (dst_bounds[2] - dst_bounds[0]) / (tile_bounds[2] - tile_bounds[0])
+                * (dst_bounds[3] - dst_bounds[1]) / (tile_bounds[3] - tile_bounds[1])
+            )
+
+            if cover_ratio < 0.01:
+                raise exceptions.TileOutOfBoundsError('dataset covers less than 1% of tile')
+
+            # compute suggested resolution in target CRS
+            dst_transform, _, _ = warp.calculate_default_transform(
                 src.crs, cls._TARGET_CRS, src.width, src.height, *src.bounds
             )
             dst_res = (abs(dst_transform.a), abs(dst_transform.e))
-            dst_bounds = warp.transform_bounds(src.crs, cls._TARGET_CRS, *src.bounds)
 
-            if bounds is None:
-                bounds = dst_bounds
+            # make sure VRT resolves the entire tile
+            tile_transform = transform.from_bounds(*tile_bounds, *tile_size)
+            tile_res = (abs(tile_transform.a), abs(tile_transform.e))
+
+            if any(tile_res < dst_res):
+                dst_res = tile_res
+                resampling_enum = cls._get_resampling_enum('nearest')
```

Comment:
@dionhaefner this is to make sure you are no oversampling right ?

Reply:
In that case, the VRT already has the exact shape and resolution required for the tile, so any further resampling is unnecessary. 

(assuming you mean setting resampling to nearest)

Reply:
To be clear, this is a minor performance optimization. Results should be identical.